lib/styled.js

```diff
@@ -5,23 +5,46 @@ const uuid = require('tiny-uuid4');
 const generateStyleString = require('./generateStyleString');
 const cache = require('./cache');
 
-function styledParser(tagName, styles, ...templateEntries) {
+function styledParser(tag, styles, ...templateEntries) {
     const styleId = uuid();
     cache.set(styleId, generateStyleString(styles, ...templateEntries));
 
-    return () => React.createElement(tagName, { className: styleId });
+    if (typeof tag === 'string') {
+        return () => React.createElement(tag, { className: styleId });
+    } else if (typeof tag === 'function') {
+        const name = tag.displayName || tag.name || 'Component';
+        return () => React.createElement(name, { className: styleId });
+    }
+
+    throw new Error('Invalid element type: expected string or class/function');
 }
 
-const styled = {};
+const styledByTags = {};
 
 htmlTagNames.forEach((htmlTagName) => {
-    styled[htmlTagName] = (styles, ...templateEntries) => styledParser(htmlTagName, styles, ...templateEntries);
+    styledByTags[htmlTagName] = (styles, ...templateEntries) => styledParser(htmlTagName, styles, ...templateEntries);
 });
 
 svgTagNames.forEach((svgTagName) => {
-    styled[svgTagName] = (styles, ...templateEntries) => styledParser(svgTagName, styles, ...templateEntries);
+    styledByTags[svgTagName] = (styles, ...templateEntries) => styledParser(svgTagName, styles, ...templateEntries);
 });
 
-Object.keys(styled).forEach((key) => { module.exports[key] = styled[key]; });
+function Styled() {
+    const styleThis = (itemToStyle) => {
+        if (typeof itemToStyle === 'string') {
+            return (...args) => styledByTags[itemToStyle](...args);
+        } else if (typeof itemToStyle === 'function') {
+            return (...args) => {
+                return styledParser(itemToStyle, ...args);
+            };
+        }
+
+        throw new Error('Invalid element type: expected string or class/function');
+    };
+
+    Object.keys(styledByTags).forEach((key) => { styleThis[key] = styledByTags[key]; });
+
+    return styleThis;
```

Comment:
Whew, getting this to work was more complex than I thought.
This way, I can call `styled()` directly, or reference a method on the object such as `styled.span` that is a template tag function.